Source/santad/EventProviders/SNTEndpointSecurityTamperResistance.mm

```diff
@@ -85,6 +85,16 @@ - (void)handleMessage:(Message &&)esMsg
       break;
     }
 
+    case ES_EVENT_TYPE_AUTH_SIGNAL: {
+      if (esMsg->event.signal.target->session_id != getpid()) {
+        result = ES_AUTH_RESULT_ALLOW;
+        break;
+      }
+      LOGW(@"Preventing attempt to kill Santa daemon");
+      result = ES_AUTH_RESULT_DENY;
+      break;
```

Comment:
Probably need a check to see if the signal's coming from launchd. It's not perfect but otherwise won't this prevent us from doing easy upgrades.

Reply:
Upgrades seem to work fine as-is (the unload is handled by the sysext framework) but this does cause a delay during reboot/shutdown, which I'm investigating workarounds for.

Reply:
Updated approach so this no longer applies.